File: core/src/main/java/com/lomekwi/cave/ui/editpanel/settings/EntryNode.java
package com.lomekwi.cave.ui.editpanel.settings;

import com.kotcrab.vis.ui.widget.VisLabel;
import com.kotcrab.vis.ui.widget.VisTree;

import java.util.function.Supplier;

public class EntryNode extends VisTree.Node<EntryNode, EntryTable, VisLabel> {
    private final Supplier<EntryTable> supplier;

    public EntryNode(String name, Supplier<EntryTable> supplier) {
        super(new VisLabel(name));
        this.supplier = supplier;
    }

    @Override
    public EntryTable getValue() {
        return supplier.get();
    }
}

File: core/src/main/scala/com/lomekwi/cave/ui/settings/EntryNode.scala
package com.lomekwi.cave.ui.settings

import com.badlogic.gdx.scenes.scene2d.ui.Tree
import com.kotcrab.vis.ui.widget.VisLabel

import java.util.function.Supplier

class EntryNode(name: String, private val supplier: Supplier[EntryTable]) extends Tree.Node[EntryNode, EntryTable, VisLabel](new VisLabel(name)) {
  override def getValue(): EntryTable = {
    supplier.get()
  }
}
